### PR TITLE
ddl: Create temporary table like tbl_xx

### LINKED ISSUE #25890: ddl: Create temporary table like tbl_xx
A subtask of #24169.

### PR DESCRIPTION
**Pull request by zhaoxugang**
### What problem does this PR solve?

Issue Number: close #25890 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- Create temporary table like tbl_xx

### CODE REVIEW

File: ddl/serial_test.go [hunk 1]
```diff
@@ -646,6 +646,28 @@ func (s *testSerialSuite) TestCreateTableWithLikeAtTemporaryMode(c *C) {
...
+	tk.MustExec("create temporary table tb13 (i int primary key, j int)")
+	_, err = tk.Exec("create temporary table tb14 like tb13;")
+	c.Assert(err.Error(), Equals, core.ErrOptOnTemporaryTable.GenWithStackByArgs("create table like").Error())
+	defer tk.MustExec("drop table if exists tb13, tb14")
+	// Test from->local temporary, to->normal
+	tk.MustExec("drop table if exists tb15, tb16")
+	tk.MustExec("create temporary table tb15 (i int primary key, j int)")
+	_, err = tk.Exec("create table tb16 like tb15;")
+	c.Assert(err.Error(), Equals, core.ErrOptOnTemporaryTable.GenWithStackByArgs("create table like").Error())
+	defer tk.MustExec("drop table if exists tb15, tb16")
```
Comment (lcwangchao) [inline ddl/serial_test.go:670]: You can add more tests to check create table like options (table_pre_split, shard_row_id_bits ...) as #25082 did

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -1708,6 +1708,30 @@ func BuildTableInfoWithCheck(ctx sessionctx.Context, s *ast.CreateTableStmt, dbC
 	return tbInfo, nil
 }
 
+// BuildSessionTemporaryTableInfo builds model.TableInfo from a SQL statement.
+func BuildSessionTemporaryTableInfo(ctx sessionctx.Context, is infoschema.InfoSchema, s *ast.CreateTableStmt, dbCharset, dbCollate string) (*model.TableInfo, error) {
```
Comment (djshow832) [inline ddl/ddl_api.go:1712]: The procedure is much less than `ddl.CreateTable`. Does it check all the things, such as 'if not exists'?

File: ddl/serial_test.go (hunk 1 above)
Comment (zhaoxugang) [reply ddl/serial_test.go:673]: done

File: ddl/ddl_api.go (hunk 2 above)
Comment (zhaoxugang) [reply ddl/ddl_api.go:1712]: yes, The procedure do it

**Review by lcwangchao — APPROVED** [review]

**Review by djshow832 — APPROVED** [review]

**Pull request merged**